* Improve style of top nav search input

PikachuEXE opened the pull request:
**Pull Request Type**
Please select what type of pull request this is:
- [x] Feature Implementation

**Related issue**
Closes #1387

**Description**
Update top nav search to add clear search input text button
Also update the padding of input text box so that input text won't appear under the "arrow-right" button

**Screenshots (if appropriate)**
https://user-images.githubusercontent.com/1018543/120764623-ebd83900-c54a-11eb-95da-a50112ef01a7.mp4
![image](https://user-images.githubusercontent.com/1018543/120764831-26da6c80-c54b-11eb-9fac-e67fc5a8f638.png)


**Testing (for code that is not small enough to be easily understandable)**
- Enter text very long into input text box
- Press the new button

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 10.15.7
 - FreeTube version: 13.0.0 (latest on `development` actually)

**Additional context**
Add any other context about the problem here.

efb4f5ff-1298-471a-8973-3d47447115dc commented:
By the way u have to add the word "closes" before the issue number to link the issue. After this PR is merged the issue will close automatically instead of staying open and clutter the backlog.

PikachuEXE commented:
I know it already
I do want a feature to auto post a comment on linked issue after PR merged instead of closing it
Sometimes an issue is not meant to be fixed completely by one PR
I will add "closes" since there is no such feature yet -,-"

PikachuEXE commented:
![image](https://user-images.githubusercontent.com/1018543/131280915-5384077c-5037-44fd-be8e-908c990dc33f.png)
Oops

PikachuEXE commented:
Issue shown in my previous comment fixed
I still got no idea how to test the look in RTL

ChunkyProgrammer commented:
I think it'd be best if this gets merged first then i can update it to be rtl compatible in my pr

peepopoggers commented:
@PikachuEXE has the placement changed?

jasonhenriquez commented on src/renderer/components/ft-input/ft-input.vue:
```diff
@@ -18,6 +20,12 @@
         :tooltip="tooltip"
       />
     </label>
+    <font-awesome-icon
+      v-if="showClearTextButton"
+      icon="backspace"
+      class="clearInputTextButton"
+      @click="handleClearTextClick"
+    />
```
```suggestion
    <font-awesome-icon
      v-if="showClearTextButton"
      icon="backspace"
      class="clearInputTextButton"
      tabindex="0"
      role="button"
      title="$t('Clear Input')"
      @click="handleClearTextClick"
      @keydown.space.prevent="handleClearTextClick"
      @keydown.enter.prevent="handleClearTextClick"
     />
```
This is inaccessible. I suggest you implement these changes here and create a yaml string "Clear Input" as well to use here as the title.

jasonhenriquez reviewed (changes requested):
I have not tested myself yet but can preliminarily see one accessibility flaw and a need for a new yaml string.

PikachuEXE commented:
Fixed issue raised by @jasonhenriquez
Updated icon to [times-circle](https://fontawesome.com/v5.15/icons/times-circle?style=solid)


https://user-images.githubusercontent.com/1018543/131792332-e5648f58-9d21-4f70-87b3-f38b730f9aa8.mov

jasonhenriquez commented:
Looks good accessibility-wise. But visually looks odd position-wise. It needs to be positioned with more space to its left.

PikachuEXE commented:
![Screenshot 2021-09-02 at 3 18 36 PM](https://user-images.githubusercontent.com/1018543/131799927-13d45264-2cf1-40d0-a3d5-2ac6e5ec8059.png)
Updated again
What do you think?

jasonhenriquez commented:
I think it would be best if the icon was a bit more right such that its hover radius was touching the text like the arrow icon:
![pr-hover](https://user-images.githubusercontent.com/84899178/131801137-9145c0d9-6f50-414b-81f8-21a3aa6d5479.png)

PikachuEXE commented:
Done
Now using `2em` as `padding-left`, same as `padding-right`

![Screenshot 2021-09-02 at 3 30 30 PM](https://user-images.githubusercontent.com/1018543/131801584-499798b5-e2b5-46e1-b3c8-7cd6328a4491.png)
![Screenshot 2021-09-02 at 3 30 35 PM](https://user-images.githubusercontent.com/1018543/131801592-a000eae3-8f97-4e60-a043-02a006dcdba7.png)
![Screenshot 2021-09-02 at 3 30 39 PM](https://user-images.githubusercontent.com/1018543/131801593-18615b39-a113-4485-b96d-11cf0563b8fa.png)

peepopoggers reviewed (changes requested):
I think the button should dissapear when the search input is empty.

Also the placement of the button is strange, on the left of the -> would make more sense to me.

And also as I said before an icon with just a cross would look much better because the -> is not in a circle so it breaks the consistency.

peepopoggers commented:
![image](https://user-images.githubusercontent.com/72892531/131897686-5d9243dc-22e5-4653-ac7c-dc6f7c4110e9.png)
What I think would make more sense placement-wise.

Using an icon [like this](https://fontawesome.com/v5.15/icons/times?style=solid)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
I disagree 

> Best spot to put it for the current UI would be on the left side of the bar instead of next to the search arrow and filter buttons as that would start to make things cluttered there and would prevent accidental clicks that clear your current search.

And i agree with this. You have to remember people that use smartphones also use FreeTube. U would definitely miss click if its next to each other.

peepopoggers commented:
@efb4f5ff-1298-471a-8973-3d47447115dc
People that use phones will use the enter key on their keyboard most likely instead of the x button

PikachuEXE commented:
@peepopoggers 
As a mobile app user, I can use the app with
1. Having the clear button on the right of text input box, **replacing** the search button (but I think there are people dislike removing existing buttons)
2. Having the clear button on the left of text input box (current)

But I cannot use the app with
1. Having the clear button on the left of search button (mis-clicking happens so often on mobile devices)

As a desktop app user, I can use the app with
1. Having the clear button on the left of text input box (current)

But I cannot use the app with
1. Having the clear button on the right of text input box, **replacing** the search button (I **need** the search button on desktop, instead of pressing `enter`)
2. Having the clear button on the left of search button (mis-clicking issue again)

PikachuEXE commented:
> the button should disappear when the search input is empty

I think it looks strange.
But you all think this is ok I will push the change.

![Screenshot 2021-09-03 at 11 07 39 AM](https://user-images.githubusercontent.com/1018543/131944741-5570dfe8-7a1b-4639-b5e6-2cc9986df1b9.png)
![Screenshot 2021-09-03 at 11 07 41 AM](https://user-images.githubusercontent.com/1018543/131944746-06adace3-3e56-43f2-a421-f168c408d6fe.png)

jasonhenriquez commented:
I can't see the changes. Did you push the commits?

PikachuEXE commented:
I excluded the change for "making the clear button when search input is empty"
Unless that change is blocking this PR from merging, I would like to leave the discussion for that design decision to another PR and make this PR merged first.

ChunkyProgrammer reviewed (dismissed):
LGTM!

jasonhenriquez commented:
I only have one more suggestion: shorten the radius of the hover effect. I do not like that it obscures the mouse "typing" cursor when focus is in the search box.

PikachuEXE commented:
Done
![image](https://user-images.githubusercontent.com/1018543/132155152-a0998104-44df-4a61-a25a-205955be10e3.png)

jasonhenriquez reviewed (approved):
LGTM!

efb4f5ff-1298-471a-8973-3d47447115dc commented:
I think merge getting blocked by @peepopoggers requested changes.

peepopoggers commented:
@efb4f5ff-1298-471a-8973-3d47447115dc fixed :)

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Yeah i think its the lint that's blocking it

efb4f5ff-1298-471a-8973-3d47447115dc commented:
> Add an X button that will clear and select the search bar in one action/click.

@PikachuEXE 
I think we missed this part. At least in my build it doesn't select the search bar after clearing the text.

jasonhenriquez commented:
This warrants a secondary PR that redirects focus and makes the button disappear after clicking.

PikachuEXE commented:
#1669 created for auto focus
"makes the button disappear after clicking" < I am not sure if this is a good UX
I am skipping it for now

efb4f5ff-1298-471a-8973-3d47447115dc commented:
Sounds logical to me to not show the button if there is no text to erase. Not sure about other implications.

PikachuEXE commented:
Here is a preview for "makes the button disappear when input text box empty"
https://user-images.githubusercontent.com/1018543/132455351-b959d89e-28b2-438f-aa34-3a698f1ce0db.mov

peepopoggers commented:
As i said before im still not a fan of the icon and placement but i guess thats preference ;/

jasonhenriquez commented:
> Here is a preview for "makes the button disappear when input text box empty"
> https://user-images.githubusercontent.com/1018543/132455351-b959d89e-28b2-438f-aa34-3a698f1ce0db.mov

I think this would look nice if it was slower / smoother. Maybe with a css animation.

PikachuEXE commented:
@jasonhenriquez 
Would you mind describing the css animation you are looking for?

jasonhenriquez commented:
A fade out with a very brief timer.

PikachuEXE commented:
@jasonhenriquez 
Please see if this is what you expect:
https://user-images.githubusercontent.com/1018543/132652935-5d4bc1f9-7066-409a-812e-090d3f1d49e2.mov

peepopoggers commented:
looks pretty cool actually

PikachuEXE commented:
Pushed all changes to #1669 
Lets continue the discussion there
